### PR TITLE
Move away from react-tools and to Babel and Istanbul.

### PR DESCRIPTION
**Pull request by joestump**
#### What does this PR do? (please provide any background)

This PR migrates away from `react-tools` to [Babel](http://babeljs.io/docs/setup/#mocha). The reason for this is because [it has all been deprecated by the React team](https://facebook.github.io/react/blog/2015/06/12/deprecating-jstransform-and-react-tools.html). I had to do [this work in another PR](https://github.com/holidayextras/tripapp-product-hotels/pull/2) for the App Platform team.

#### What tests does this PR have?

No tests, but I did run them locally and they are passing.

#### How can be this tested?

```
npm install
npm run lint && npm test
```

#### Screenshots / Screencast
n/a

#### What gif best describes how you feel about this work?

---
#### Developer Definition of Done/Quality Checklist (for PR author to complete BEFORE code review):
- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md) and I'm happy for this to be reviewed.

#### Software Engineer or Developer review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve run all the tests locally and they pass.

#### Software Engineer or project guru review:
- [x] I’ve witnessed the work behaving as expected (this could be on the authors machine or screencast).
- [x] I’ve checked for coding anti-patterns.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve run all the tests locally and they pass.

### CODE REVIEW

**Comment by joestump** [comment]
@jackdcrawford this should now be passing tests, but the source view on Coveralls is not working. It appears to be a common problem with absolute paths. Gonna see if CI magically works.

**Comment by joestump** [comment]
@jackdcrawford & @jodiedoubleday Okay this is all working again and ready for a review.

**Comment by hxoliverrumbelow** [comment]
I'd like to know the driving force behind moving from `blanket` to `istanbul`, is it purely for the babel transformation step, or is one superior to the other?

**Comment by jodiedoubleday** [comment]
Also what are the advantages of using Chai here? It doesn't look like you are using any of the features from it. What are the advantage of using it?

File: tests/bootstrap.js [hunk 1]
```diff
@@ -1,11 +1,6 @@
-// Common test utilities.
-global.assert = require('chai').assert;
```
Comment (joestump) [inline tests/bootstrap.js]: @jodiedoubleday we've always used `assert` from `chai`. All I did was nuke it from `window`.

**Comment by joestump** [comment]
@hxoliverrumbelow I moved to Istanbul for a number of reasons:

* When I was poking around, getting Babel up and running with Istanbul was faster and more tightly integrated. Blanket still required JSX transformers, etc. 
* [Blanket 2.0 is moving to Istanbul under the hood](https://github.com/alex-seville/blanket/issues/310) with the author stating it's because, "I think this will allow Blanket to leverage the advanced features of Istanbul, and provide better value to users of both libraries."

Istanbul appears to be both more widely supported/used as well as more robust. It was also the easiest and cleanest to get working.

**Comment by Joezo** [comment]
Took the dev review, plus 1 :+1: :sparkles:

**Comment by robhuzzey** [comment]
Taking SE/GURU

**Comment by robhuzzey** [comment]
:+1: from me

**Pull request merged**